core/src/main/java/ai/timefold/solver/core/impl/domain/variable/cascade/CascadingUpdateShadowVariableDescriptor.java

```diff
@@ -118,20 +119,24 @@ public void linkVariableDescriptors(DescriptorPolicy descriptorPolicy) {
                 .orElse(null);
 
         var targetMethodName = getTargetMethodName();
-        var sourceMethodMember = ConfigUtils.getDeclaredMembers(entityDescriptor.getEntityClass())
+        var sourceMethodMembers = ConfigUtils.getDeclaredMembers(entityDescriptor.getEntityClass())
                 .stream()
                 .filter(member -> member.getName().equals(targetMethodName))
-                .findFirst()
-                .orElse(null);
-        if (sourceMethodMember == null) {
+                .toList();
+        if (sourceMethodMembers.size() > 1) {
+            throw new IllegalArgumentException("""
+                    The entity class (%s) has multiple members named as "%s".
+                    Maybe rename the method "%s" with a unique name.""".formatted(entityDescriptor.getEntityClass(),
+                    targetMethodName, targetMethodName));
+        } else if (sourceMethodMembers.stream().noneMatch(m -> Method.class.isAssignableFrom(m.getClass()))) {
```

Comment:
I would honestly do it like this:

```java
 var sourceMethodMembers = ConfigUtils.getDeclaredMembers(entityDescriptor.getEntityClass())
                .stream()
                .filter(member -> member instanceof Method method && 
                        method.getName().equals(targetMethod) && 
                        method.getParameterCount() == 0)
                .toList();
```

This:

- Select only 0-arg methods with the given name, which are the only valid choice

Since only 0-arg methods would be selected, it would be impossible to choose a field accidentally, and we can allow having a field with the same name as a method.

This also makes the `sourceMethodMembers.stream().noneMatch(m -> Method.class.isAssignableFrom(m.getClass()))` never trigger. In normal classes, `sourceMethodMembers.size() > 1` would also never trigger, but generated classes allow multiple 0-arg methods with the same name (provided they have a different return type).

Reply:
I see your point, but in that case, we won't fail if there is a field and a method using the same name.

Reply:
And why should we fail? Unlike variables (which are either fields or a getter/setter pair, and thus ambiguity arises if both a field and its getter are annotated with PlanningVariable), this is always a method and never a field (it even has method in the name), so having a field and method with the same name does not create ambiguity.